<launch>
    <arg name="pluginlists_yaml" default="$(find mavros)/launch/px4_pluginlists.yaml" />
    <arg name="config_yaml" default="$(find mavros)/launch/px4_config.yaml" />

    <group ns="uav1">
        <!--        <arg name="fcu_url" default="/dev/MasterUAV:921600"/>-->
<!--        <arg name="fcu_url" default="/dev/NoopS0:921600"/>-->
        <arg name="fcu_url" default="/dev/CUAVM0:230400"/>
        <arg name="gcs_url" value="udp://:14560@localhost:14550"/>
        <arg name="tgt_system" value="1"/>
        <arg name="tgt_component" value="1"/>
        <arg name="log_output" default="screen" />

        <arg name="fcu_protocol" default="v2.0" />
        <arg name="respawn_mavros" default="false" />

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(arg pluginlists_yaml)" />
            <arg name="config_yaml" value="$(arg config_yaml)" />

            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)" />
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)" />
            <arg name="log_output" value="$(arg log_output)" />
            <arg name="fcu_protocol" value="$(arg fcu_protocol)" />
            <arg name="respawn_mavros" default="$(arg respawn_mavros)" />
        </include>
    </group>

    <!-- UAV2 iris_2 -->
    <group ns="uav2">
<!--        <arg name="fcu_url" default="/dev/CUAVM1:230400"/>-->
<!--        <arg name="fcu_url" default="/dev/REMOTE915_SC2:115200"/>-->
        <arg name="fcu_url" default="/dev/NoopS0:921600"/>
        <arg name="gcs_url" value="udp://:14561@localhost:14550"/>
        <arg name="tgt_system" value="2"/>
        <arg name="tgt_component" value="1"/>
        <arg name="log_output" default="screen" />
        <arg name="fcu_protocol" default="v2.0" />
        <arg name="respawn_mavros" default="false" />

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(arg pluginlists_yaml)" />
            <arg name="config_yaml" value="$(arg config_yaml)" />

            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)" />
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)" />
            <arg name="log_output" value="$(arg log_output)" />
            <arg name="fcu_protocol" value="$(arg fcu_protocol)" />
            <arg name="respawn_mavros" default="$(arg respawn_mavros)" />
        </include>
    </group>

    <group ns="uav3">
        <arg name="fcu_url" default="/dev/NoopS2:921600"/>
        <arg name="gcs_url" value="udp://:14562@localhost:14550"/>
        <arg name="tgt_system" value="3"/>
        <arg name="tgt_component" value="1"/>
        <arg name="log_output" default="screen" />
        <arg name="fcu_protocol" default="v2.0" />
        <arg name="respawn_mavros" default="false" />

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(arg pluginlists_yaml)" />
            <arg name="config_yaml" value="$(arg config_yaml)" />

            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)" />
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)" />
            <arg name="log_output" value="$(arg log_output)" />
            <arg name="fcu_protocol" value="$(arg fcu_protocol)" />
            <arg name="respawn_mavros" default="$(arg respawn_mavros)" />
        </include>
    </group>

    <group ns="uav4">
        <arg name="fcu_url" default="/dev/433UAV4:115200"/>
        <arg name="gcs_url" value="udp://:14563@localhost:14550"/>
        <arg name="tgt_system" value="4"/>
        <arg name="tgt_component" value="1"/>
        <arg name="log_output" default="screen" />
        <arg name="fcu_protocol" default="v2.0" />
        <arg name="respawn_mavros" default="false" />

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(arg pluginlists_yaml)" />
            <arg name="config_yaml" value="$(arg config_yaml)" />

            <arg name="fcu_url" value="$(arg fcu_url)" />
            <arg name="gcs_url" value="$(arg gcs_url)" />
            <arg name="tgt_system" value="$(arg tgt_system)" />
            <arg name="tgt_component" value="$(arg tgt_component)" />
            <arg name="log_output" value="$(arg log_output)" />
            <arg name="fcu_protocol" value="$(arg fcu_protocol)" />
            <arg name="respawn_mavros" default="$(arg respawn_mavros)" />
        </include>
    </group>

    <include file="$(find offboard)/launch/rosbag_log.launch"/>
    <arg name="formation_distance" default="10" />
    <node pkg = "offboard" name = "offboard" type="4uav" output="screen">
        <param name="formation_distance" value="$(arg formation_distance)" />
    </node>
</launch>